scipy/sparse/_construct.py

```diff
@@ -70,39 +72,32 @@ def spdiags(data, diags, m=None, n=None, format=None):
     return dia_matrix((data, diags), shape=(m, n)).asformat(format)
 
 
-def diags(diagonals, offsets=0, shape=None, format=None, dtype=None):
+def diags_array(diagonals, /, offsets=0, *, shape=None, format=None, dtype=None):
```

Comment:
Making `offsets` keyword-only would make sense to me. `diags_array(([0, 2, 7], [1, 3, 9]), (1, 2))` is a lot more confusing than `diags_array(([0, 2, 7], [1, 3, 9]), offsets=(1, 2))`

Reply:
I used the test suite to determine where the `*` should go; if you make `offsets` keyword-only then the `test_diags_*` tests start failing. It's a simple-enough thing to fix up, but I had treated the failing tests as a proxy for "in-the-wild" usage.

Of course, this is a new function so we can introduce new constraints - I'm happy to do so if that's what's preferred!

Reply:
After some discussion with @perimosocordiae the consensus is that making `offsets` keyword-only as well would be an improvement, so I will go ahead and make that change. Thanks for the suggestion @rgommers !